src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/MultiFileCompilationModuleGroup.cs

```diff
@@ -101,7 +101,7 @@ public MultiFileSharedCompilationModuleGroup(CompilerTypeSystemContext context,
 
         public override bool ShouldProduceFullVTable(TypeDesc type)
         {
-            return ConstructedEETypeNode.CreationAllowed(type);
+            return false;
```

Comment:
Could you make a copy of this class that will be local to the test infrastructure and make the test runner use that?

This class is used for multifile mode and this would break it. We don't test it currently because of some linker issue (https://github.com/dotnet/runtime/issues/66191).